napi-inl.h

```diff
@@ -2685,8 +2680,14 @@ inline AsyncWorker::AsyncWorker(const Object& receiver, const Function& callback
   : _env(callback.Env()),
     _receiver(Napi::Persistent(receiver)),
     _callback(Napi::Persistent(callback)) {
-  napi_status status = napi_create_async_work(
-    _env, OnExecute, OnWorkComplete, this, &_work);
+
+  napi_value resource_id;
+  napi_status status = napi_create_string_latin1(
+    _env, "generic", -1, &resource_id);
```

Comment:
I don't like this much, but it's fine for now. The follow-up changes will move this resource name string to a parameter of the `AsyncWorker` constructor.